File: Paiza/Python/Aランクレベルアップメニュー/最大公約数/素数判定.py
import math
n = int(input())

a = math.floor(math.sqrt(n))+1
factor = 0
for i in range(2, a):
    if(n % i == 0):
        factor +=1

if factor == 0 and n > 1:
    print('YES')
else:
    print('NO')

File: C#/素数判定.cs


//　素数判定

    public static bool IsPrime(double num)
        {
            if (num % 2 == 0) return false;
        
            double sqrtNum = Math.Sqrt(num);
            for (int i = 3; i <= sqrtNum; i += 2)
            {
                if (num % i == 0)
                {
                    return false;
                }
            }
        
            return true;
        }